### ISSUE TITLE
Accessibility: Visual label missing

### ISSUE THREAD

**Issue by vikrampm1** [woothemes/woocommerce#51930]
**Prerequisites (mark completed items with an [x]):**
- [x] I have checked that my issue type is not listed here https://github.com/woocommerce/woocommerce/issues/new/choose
- [x] My issue is not a security issue, support request, bug report, enhancement or feature request (Please use the link above if it is).

### Issue Description:

There is no visual lable for the sorting dropdown. Some users may not understand what this dropdown is for.

![Image](https://github.com/user-attachments/assets/c02541c7-2268-48a2-ad10-644a82749dfc)

### Component or Page: 

Component: Product Archive "Sort By" Block

### Name affected element:

"Shop order Default sorting" combobox

### Relevant code snippets:

```
<div data-block-name=""woocommerce/catalog-sorting"" class=""woocommerce wc-block-catalog-sorting has-font-size has-small-font-size "" style=""""><form class=""woocommerce-ordering"" method=""get"">
	<select name=""orderby"" class=""orderby"" aria-label=""Shop order"">
					<option value=""menu_order"" selected=""selected"">Default sorting</option>
					<option value=""popularity"">Sort by popularity</option>
					<option value=""rating"">Sort by average rating</option>
					<option value=""date"">Sort by latest</option>
					<option value=""price"">Sort by price: low to high</option>
					<option value=""price-desc"">Sort by price: high to low</option>
			</select>
	<input type=""hidden"" name=""paged"" value=""1"">
	</form>
</div>
```

### Recommended fix:

Provide a visible label for the input that says ""Sort by: "". Most ecommerce stores place this lable to the left of the dropdown. 

Here's an examples of label placement to the left of the dropdown on the BestBuy website: https://drive.google.com/file/d/1WAkw48xxhnfrv5tYx3lUSRP84TGMYlwl/view?usp=drive_link

**Comment by Manussakis** [comment]
The sorting dropdown doesn't have a visual label decoupled from the input value because the value itself works as a visual label.

A similar technique is used on the block-based Checkout page when the inputs are empty, for example.

![Image](https://github.com/user-attachments/assets/1ca19cc0-8eb8-498d-b467-883e8d933c64)

The "First name" and "Last name" placeholders work as visual labels. When these inputs get content, the label floats and both the label and value are visible.

![Image](https://github.com/user-attachments/assets/7cff22c7-302a-432f-9ad1-6f320a121661)

The only difference is that for the sorting input, both the label and the value have the same content and there is no need to repeat them. 

For screen-reader users, the sorting dropdown already has the `aria-label` attribute that reads "Shop order" which provides users with information on what the dropdown is for.

### There is an alternative, but...

An alternative would be having the "Sort by:" text as part of the label and the sorting dropdown with the options' text like: "Popularity", "Average rating", "Latest", etc.

![Image](https://github.com/user-attachments/assets/76fcc466-7957-4e06-b4f7-9b7a539a30f2)

The problem is that these values pass through the [woocommerce_catalog_orderby](https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce/includes/wc-template-functions.php#L1550) filter and modifying it is prone to regressions, as other plugins and themes may rely on it.

Since the lack of a visible label separate from the select input does not prevent users from sorting products and given that a refactor could impact other themes and plugins, I suggest sticking to the current solution and closing this issue.


**Reference**:
- [Label in Name (Level A)](https://www.w3.org/WAI/WCAG21/Understanding/label-in-name.html)
- [Visible label is part of accessible name](https://www.w3.org/WAI/standards-guidelines/act/rules/2ee8b8/proposed/)

**Similar solution**
- Amazon
![Image](https://github.com/user-attachments/assets/df71adcf-1df3-4819-9c4d-c922e77a254b)

cc: @vikrampm1

**Comment by vikrampm1** [comment]
@amberhinds any objection on closing this issue as per the comment above?

**Comment by amberhinds** [comment]
No, this is a WCAG failure and should not be closed.

For sighted people, it would be most ideal to have the label separate. Also, there are other components that do include the label to the left, such as the order by select on the reviews block.

![Image](https://github.com/user-attachments/assets/202b2c68-746d-4142-9aa4-92b6c5e81e28)

The current implementation does not pass [Label in Name (Level A)](https://www.w3.org/WAI/WCAG21/Understanding/label-in-name.html) because the visible label is **not** part of the accessible name.

I.e., the accessible name on this select is "Shop Order", which is what the aria-label is set to:

```
<select name="orderby" class="orderby" aria-label="Shop order">
					<option value="menu_order" selected="selected">Default sorting</option>
					<option value="popularity">Sort by popularity</option>
					<option value="rating">Sort by average rating</option>
					<option value="date">Sort by latest</option>
					<option value="price">Sort by price: low to high</option>
					<option value="price-desc">Sort by price: high to low</option>
			</select>
```

Label in Name requires that the text of the label (or visible text) be at the **start** of the accessible name. For example:
- ✅ Pass: a link with visual text that says "Learn More" and an aria-label that says "Learn more about us"
- ❌ Fail: a a link with visual text that says "Learn More" and an aria-label that says "Click here to Learn more about us"

This would also fail Label in Name unless the aria-label included the visual text at the start, such as "Default sorting—shop order." The problem with this is that you would have to change the aria-label every time the visual text changes, which would cause a problem for screen reader users.

In its current implementation, it would be incredibly difficult for someone who relies on voice control to know what to say to be able to select the element.

The All Products block also fails on this front as the label is visually hidden, though it is present.

```
<div class="wc-block-sort-select wc-block-components-sort-select wc-block-product-sort-select wc-block-components-product-sort-select">
     <label class="wc-block-sort-select__label wc-block-components-sort-select__label" for="wc-block-components-sort-select__select-0"><span aria-hidden="true"></span><span class="screen-reader-text">Order products by</span></label>
     <select id="wc-block-components-sort-select__select-0" class="wc-block-sort-select__select wc-block-components-sort-select__select">
               <option value="menu_order">Default sorting</option>
               <option value="popularity">Popularity</option>
               <option value="rating">Average rating</option>
               <option value="date">Latest</option>
               <option value="price">Price: low to high</option>
               <option value="price-desc">Price: high to low</option>
     </select>
</div>
```

Would it be possible to reuse what has been implemented in the All Products block and remove the screen reader only class from the label for both?

**Comment by amberhinds** [comment]
Also, as an FYI, @Manussakis, the block-based checkout does have labels. They are styled to look like placeholder text, but they are actual labels. 😀

![Image](https://github.com/user-attachments/assets/19f99426-824c-46b2-9d63-1e8b35d340fe)

```
<div class="wc-block-components-text-input wc-block-components-address-form__first_name has-error">
     <input type="text" id="shipping-first_name" autocapitalize="sentences" autocomplete="given-name" aria-label="First name" required="" aria-invalid="true" title="" value="" aria-describedby="validate-error-shipping_first_name">
     <label for="shipping-first_name">First name</label>
     <div class="wc-block-components-validation-error" role="alert"><p>Please enter a valid first name</p></div>
</div>
```

**Comment by Manussakis** [comment]
@amberhinds Thanks for the detailed explanation!
I will go for this approach, as flagged in my initial comment
![Image](https://github.com/user-attachments/assets/9a057206-d814-4ff3-ac4e-86987fed087c)
and coordinate with the Woo team on how to mitigate the possible regressions due to changing the returned value of the [woocommerce_catalog_orderby](https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce/includes/wc-template-functions.php#L1550) filter.

---

> Also, as an FYI, @Manussakis, the block-based checkout does have labels. They are styled to look like placeholder text, but they are actual labels.

I do know it. I meant that in both cases there is no visual label separate from the input, but I got your point. 😉  
Thanks!

**Comment by amberhinds** [comment]
Thanks, @Manussakis. I think that's the best approach.
